src/ChannelServer/Network/Helpers/BuffHelper.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Melia.Channel.World;
+using Melia.Channel.World.Entities;
+using Melia.Shared.Network;
+
+namespace Melia.Channel.Network.Helpers
+{
+	public static class BuffHelper
+	{
+		/// <summary>
+		/// Adds buff data to packet.
+		/// </summary>
+		/// <param name="packet"></param>
+		/// <param name="buff"></param>
+		public static void AddBuff(this Packet packet, IEntity entity, Buff buff)
+		{
+			packet.PutInt(entity.Handle);
+			packet.PutInt(buff.Handle);
+			packet.PutInt((int)buff.Id);
+			packet.PutInt(1);
+			packet.PutInt(0);
+			packet.PutInt(0);
+			packet.PutInt(0);
+			packet.PutInt(0);
+			packet.PutInt(buff.OverbuffCounter);
+			packet.PutInt(buff.Data.Duration);
+			packet.PutInt(0);
+			packet.PutInt(0);
+			if (entity is Character)
```

Comment:
Currently, I wasn't too sure how the server adds buffs to monsters, needs more info, so I handled the case for characters. Rather than add un-needed fields to IEntity. An example is how the server handles monster's deaths vs player deaths
ZC_DEAD is sent with no killer info on player death while on monster death it is set.
```C#
		/// <summary>
		/// Broadcasts ZC_DEAD in range of entity, which kills it visually.
		/// </summary>
		/// <param name="entity">Entity to kill visually.</param>
		public static void ZC_DEAD(IEntity entity, IEntity killer)
		{
			var packet = new Packet(Op.ZC_DEAD);
			packet.PutInt(entity.Handle);
			packet.PutByte(0);
			if (killer != null)
				packet.PutByte(1);
			else
				packet.PutByte(0);
			packet.PutByte(0);
			packet.PutByte(0);
			packet.PutPosition(entity.Position);
			packet.PutInt(0);
			if (killer != null)
			{
				packet.PutInt(killer.Handle);
				packet.PutInt(0);
				packet.PutLong(0x40);
				packet.PutLong(0x35);
				packet.PutLong(0x40);
			}

			entity.Map.Broadcast(packet, entity);
		}
```

Reply:
I see. Though the difference in `ZC_DEAD´ could also be in the killer, instead of the dead. I guess we'd need a PvP log to say for sure.

Reply:
IEntity now has a Name property.